app/api/routers/query.py

```diff
@@ -32,7 +32,7 @@
 # For more context, see https://github.com/neurobagel/api/issues/327.
 @router.get("", response_model=List[CohortQueryResponse])
 async def get_query(
-    query: QueryModel = Depends(QueryModel),
+    query: Annotated[QueryModel, Query()],
```

Comment:
out of interest: what does `Query()` do here?

Reply:
If it's unusual even for pydantic v2, maybe a comment above this line with an explainer would be good

Reply:
This is a FastAPI feature rather than a Pydantic feature: https://fastapi.tiangolo.com/tutorial/query-param-models/. `Query()` is how you explicitly define a query parameter in FastAPI and declare things like additional validation, as used by parameters in our QueryModel:

https://github.com/neurobagel/api/blob/3718200213d81a07755bcd97365f6a5a2918dd97/app/api/models.py#L14-L28

<v0.115.0, FastAPI didn't "officially" support Pydantic models as query parameters, hence the old dependency syntax we were using (which was the documented way to use a data model at the time).